apps/wallet/src/ui/app/shared/faucet/slice.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { createSlice } from '@reduxjs/toolkit';
+
+import { requestGas } from './actions';
+
+type SliceType = {
+    loading: boolean;
+    lastRequest: {
+        status: number;
+        statusTxt: string;
+        error: boolean;
+        totalGasReceived?: number;
+        retryAfter?: number;
+    } | null;
+};
+
+const initialState: SliceType = {
+    loading: false,
+    lastRequest: null,
+};
+
+const slice = createSlice({
```

Comment:
Not essential for now, but this would be a good candidate for react query. I think it'd really clean things up and simplify the logic (and avoid the middleware).

Reply:
Yeah with a `useMutation` + `react-hot-toast` you should be able to just pass the promise from the mutation to the toast and it'll even deal with transitioning the state.